cg/meta/workflow/analysis.py

```diff
@@ -0,0 +1,357 @@
+            raise CgError
+        elif not case_obj.links:
+            LOG.error("Case %s has no samples in StatusDB!", case_id)
+            raise CgError
+
+    def check_analysis_ongoing(self, case_id: str) -> None:
+        if self.trailblazer_api.is_latest_analysis_ongoing(case_id=case_id):
+            LOG.warning(f"{case_id} : analysis is still ongoing - skipping")
+            raise CgError("Analysis still ongoing")
+
+    def get_flowcells(self, case_id: str) -> List[models.Flowcell]:
+        """Get all flowcells for all samples in a ticket"""
+        flowcells = set()
+        case_obj: models.Family = self.status_db.family(case_id)
+        for familysample in case_obj.links:
+            for flowcell in familysample.sample.flowcells:
+                flowcells.add(flowcell)
+        return list(flowcells)
+
+    def all_flowcells_on_disk(self, case_id: str) -> bool:
+        """Check if flowcells are on disk for sample before starting the analysis.
+        Flowcells not on disk will be requested
+        """
+        flowcells = self.get_flowcells(case_id=case_id)
+        statuses = []
+        for flowcell_obj in flowcells:
+            LOG.debug(f"{flowcell_obj.name}: checking if flowcell is on disk")
+            statuses.append(flowcell_obj.status)
+            if flowcell_obj.status == "removed":
+                LOG.info(f"{flowcell_obj.name}: flowcell not on disk, requesting")
+                flowcell_obj.status = "requested"
+            elif flowcell_obj.status != "ondisk":
+                LOG.warning(f"{flowcell_obj.name}: {flowcell_obj.status}")
+        self.status_db.commit()
+        return all(status == "ondisk" for status in statuses)
+
+    def get_priority_for_case(self, case_id: str) -> str:
+        """Fetch priority for case id"""
+        case_obj: models.Family = self.status_db.family(case_id)
+        if not case_obj.priority or case_obj.priority == 0:
+            return "low"
+        if case_obj.priority > 1:
+            return "high"
+        return "normal"
+
+    def get_case_path(self, case_id: str) -> Path:
+        raise NotImplementedError
+
+    def get_case_config_path(self, case_id) -> Path:
+        raise NotImplementedError
+
+    def get_trailblazer_config_path(self, case_id: str) -> Path:
+        raise NotImplementedError
+
+    def get_sample_name_from_lims_id(self, lims_id: str) -> str:
+        """
+        Retrieve sample name provided by customer for specific sample
+        """
+        sample_obj: models.Sample = self.status_db.sample(lims_id)
+        return sample_obj.name
+
+    def link_fastq_files(self, case_id: str, dry_run: bool = False) -> None:
+        """
+        Links fastq files from Housekeeper to case working directory
+        """
+        raise NotImplementedError
+
+    def get_bundle_deliverables_type(self, case_id: str) -> Optional[str]:
+        return None
+
+    @staticmethod
+    def get_application_type(sample_obj: models.Sample) -> str:
```

Comment:
A dock string describing why we have this hard coded stuff here. Or maybe handle this by instead adding another collumn to the database wich would define the functionallity of the analysis_type? Or something else...